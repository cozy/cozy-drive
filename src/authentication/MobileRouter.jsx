import React from 'react'
import { Router, withRouter } from 'react-router'

import Authentication from './Authentication'
import Revoked from './Revoked'

const MobileRouter = ({
  history,
  appRoutes,
  isAuthenticated,
  isRevoked,
  onAuthenticated,
  onLogout
}) => {
  if (!isAuthenticated) {
    return <Authentication router={history} onComplete={onAuthenticated} />
  } else if (isRevoked) {
    return (
      <Revoked
        router={history}
        onLogBackIn={onAuthenticated}
        onLogout={onLogout}
      />
    )
  } else {
    return <Router history={history}>{appRoutes}</Router>
  }
}

export default withRouter(MobileRouter)
