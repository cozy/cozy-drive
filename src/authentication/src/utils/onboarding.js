import localforage from 'localforage'
import get from 'lodash/get'

const ONBOARDING_SECRET_KEY = 'onboarding_secret'
const ONBOARDING_STATE = 'onboarding_state'

const generateRandomString = () => {
  return Math.random()
    .toString(36)
    .substr(2, 11)
}
export const writeSecret = secret => {
  return localforage.setItem(ONBOARDING_SECRET_KEY, secret)
}

export const readSecret = () => {
  return localforage.getItem(ONBOARDING_SECRET_KEY)
}

export const clearSecret = () => {
  return localforage.removeItem(ONBOARDING_SECRET_KEY)
}

export const writeState = state => {
  return localforage.setItem(ONBOARDING_STATE, state)
}

export const readState = () => {
  return localforage.getItem(ONBOARDING_STATE)
}

export const clearState = () => {
  return localforage.removeItem(ONBOARDING_STATE)
}
const generateState = () => {
  return generateRandomString()
}
const generateSecret = () => {
  return generateRandomString()
}
export const checkIfOnboardingLogin = onboardingInformations => {
  return get(onboardingInformations, 'code')
}

export const generateOAuthForUrl = async ({
  clientName,
  redirectURI,
  softwareID,
  softwareVersion,
  clientURI,
  logoURI,
  policyURI,
  scope
}) => {
  let secret = await readSecret()
  if (!secret) {
    secret = generateSecret()
    await writeSecret(secret)
  }
  let state = await readState()

  if (!state) {
    state = generateState()
    await writeState(state)
  }
  const oauthData = {
    redirect_uri: redirectURI,
    software_id: softwareID,
    client_name: clientName,
    software_version: softwareVersion,
    client_kind: 'mobile',
    client_uri: clientURI,
    logo_uri: logoURI,
    policiy_uri: policyURI,
    onboarding: {
      app: softwareID,
      permissions: scope,
      secret: secret,
      state: state
    }
  }

  return encodeURIComponent(JSON.stringify(oauthData))
}

export const addProtocolToURL = instanceDomain => {
  return `https://${instanceDomain}`
}
export const secretExchange = (secret, instanceDomain, client) => {
  const response = client.stackClient.fetchJSON(
    'POST',
    addProtocolToURL(instanceDomain) + '/auth/secret_exchange',
    {
      secret
    }
  )
  return response
}

export const getAccessToken = async (
  { client_id, client_secret },
  instanceDomain,
  code,
  client
) => {
  const body = `grant_type=authorization_code&code=${code}&client_id=${client_id}&client_secret=${client_secret}`
  const getTokenRequest = await client.stackClient.fetch(
    'POST',
    addProtocolToURL(instanceDomain) + '/auth/access_token',
    body,
    {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    }
  )
  const token = await getTokenRequest.json()
  if (getTokenRequest.status !== 200) {
    throw new Error('token.error')
  }
  return token
}
