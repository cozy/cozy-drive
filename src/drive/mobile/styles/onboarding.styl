@require 'components/forms.styl'
@require 'settings/palette.styl'
@require 'settings/z-index.styl'

.wizard
    @extend $form
    @extend $form-text
    position fixed
    top 0
    left 0
    height 100%
    z-index $bar-index
    display flex
    flex-direction column
    width   85vw
    padding 0 7.5vw
    background-color white
    color            black
    text-align center

    .wizard-header
        display         flex
        flex-direction  row
        margin-top      2rem
        justify-content flex-start

        .skipLink
            padding         .5rem
            cursor          pointer
            text-transform  uppercase
            text-align      right
            color           coolGrey
            margin-left     auto

    .illustration
        margin .5rem auto
        height 14rem
        background-size contain

    .wizard-main
        margin auto
        padding 0 .5rem
        width 100%
        box-sizing border-box

    .title
        margin    .25rem auto
        font-size 1.5rem

    .description
        text-align center
        margin     .5rem auto
        font-size  1.125rem
        max-width  25rem


    .wizard-footer
        display flex
        flex-direction column
        padding-bottom env(safe-area-inset-bottom)
        margin-bottom 1.5rem


// ASSETS

.illustration-photos
    background      embedurl("../assets/illustrations/illustration-photos.svg") center no-repeat

.illustration-analytics
    background      embedurl("../assets/illustrations/illustration-trophy.svg") center no-repeat

.illustration-files
    background      embedurl("../assets/illustrations/illustration-files.svg") center no-repeat
