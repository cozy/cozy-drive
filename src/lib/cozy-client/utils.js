export const mapValues = (object, transform) => {
  let result = {}
  for (const key in object) {
    result[key] = transform(object[key], key)
  }
  return result
}

export const filterValues = (object, filter) => {
  let result = {}
  for (const key in object) {
    if (filter(object[key], key)) {
      result[key] = object[key]
    }
  }
  return result
}

export const removeObjectProperty = (obj, prop) => {
  return Object.keys(obj).reduce((result, key) => {
    if (key !== prop) {
      result[key] = obj[key]
    }
    return result
  }, {})
}

export const removeObjectProperties = (obj, props) => {
  const sProps = new Set(props)
  const res = Object.keys(obj).reduce((result, key) => {
    if (!sProps.has(key)) {
      result[key] = obj[key]
    }
    return result
  }, {})
  return res
}

export const sleep = (time, args) => {
  return new Promise(resolve => {
    setTimeout(resolve, time, args)
  })
}

export const retry = (fn, count, delay = 300) => {
  return function doTry(...args) {
    return fn(...args).catch(err => {
      if (--count < 0) {
        throw err
      }
      return sleep(getBackedoffDelay(delay, count)).then(() => doTry(...args))
    })
  }
}

const FUZZ_FACTOR = 0.3
export const getFuzzedDelay = retryDelay => {
  const fuzzingFactor = (Math.random() * 2 - 1) * FUZZ_FACTOR
  return retryDelay * (1.0 + fuzzingFactor)
}

export const getBackedoffDelay = (retryDelay, retryCount = 1) =>
  getFuzzedDelay(retryDelay * Math.pow(2, retryCount - 1))
