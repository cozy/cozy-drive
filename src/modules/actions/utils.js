import { isDirectory } from 'cozy-client/dist/models/file'
import { receiveQueryResult } from 'cozy-client/dist/store'

import { DOCTYPE_FILES } from '@/lib/doctypes'
import {
  getEncryptionKeyFromDirId,
  downloadEncryptedFile,
  isEncryptedFolder
} from '@/lib/encryption'

const isMissingFileError = error => error.status === 404

const downloadFileError = error => {
  return isMissingFileError(error)
    ? 'error.download_file.missing'
    : 'error.download_file.offline'
}

/**
 * An instance of cozy-client
 * @typedef {object} CozyClient
 */

/**
 * downloadFiles - Triggers the download of one or multiple files by the browser
 *
 * @param {CozyClient} client
 * @param {array} files  One or more files to download
 */
export const downloadFiles = async (
  client,
  files,
  { vaultClient, showAlert, t } = {},
  driveId
) => {
  const encryptionKey = await getEncryptionKeyFromDirId(client, files[0].dir_id)

  if (files.length === 1 && !isDirectory(files[0])) {
    const file = files[0]
    try {
      const filename = file.name
      if (encryptionKey) {
        return downloadEncryptedFile(client, vaultClient, {
          file,
          encryptionKey
        })
      } else {
        return client
          .collection(DOCTYPE_FILES, { driveId })
          .download(file, null, filename)
      }
    } catch (error) {
      showAlert({ message: t(downloadFileError(error)), severity: 'error' })
    }
  } else {
    if (encryptionKey) {
      // Multiple download is forbidden for encrypted files because we cannot generate client archive for now.
      return showAlert({
        message: t('error.download_file.encryption_many'),
        severity: 'error'
      })
    }
    const hasEncryptedDirs = files.find(
      file => isDirectory(file) && isEncryptedFolder(file)
    )
    if (hasEncryptedDirs) {
      // We cannot download encrypted folder because we cannot generate client archive for now.
      return showAlert({
        message: t('error.download_file.encryption_many'),
        severity: 'error'
      })
    }
    const ids = files.map(f => f.id)
    return client.collection(DOCTYPE_FILES).downloadArchive(ids)
  }
}

const isAlreadyInTrash = err => {
  const reasons = err.reason !== undefined ? err.reason.errors : undefined
  if (reasons) {
    for (const reason of reasons) {
      if (reason.detail === 'File or directory is already in the trash') {
        return true
      }
    }
  }
  return false
}

/**
 * trashFiles - Moves a set of files to the cozy trash
 *
 * @param {CozyClient} client
 * @param {array} files  One or more files to trash
 */
export const trashFiles = async (client, files, { showAlert, t }) => {
  try {
    for (const file of files) {
      // TODO we should not go through a FileCollection to destroy the file, but
      // only do client.destroy(), I do not know what it did not update the internal
      // store correctly when I tried
      const { data: updatedFile } = await client
        .collection(DOCTYPE_FILES)
        .destroy(file)
      client.store.dispatch(
        receiveQueryResult(null, {
          data: updatedFile
        })
      )
      client.collection('io.cozy.permissions').revokeSharingLink(file)
    }

    showAlert({ message: t('alert.trash_file_success'), severity: 'success' })
  } catch (err) {
    if (!isAlreadyInTrash(err)) {
      showAlert({ message: t('alert.try_again'), severity: 'error' })
    }
  }
}

export const restoreFiles = async (client, files) => {
  for (const file of files) {
    await client.collection(DOCTYPE_FILES).restore(file.id)
  }
}
