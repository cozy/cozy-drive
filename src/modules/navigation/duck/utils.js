export const getSharingIdFromUrl = url => {
  const urlSearchParams = new URLSearchParams(url.search)
  return urlSearchParams.get('sharing')
}
