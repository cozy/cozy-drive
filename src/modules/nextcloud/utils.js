export function makePath(root, filename) {
  return `${root}${root.endsWith('/') ? '' : '/'}${filename}`
}
