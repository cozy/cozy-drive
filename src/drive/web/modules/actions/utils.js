import { models } from 'cozy-client'
import { isMobileApp, isIOS } from 'cozy-device-helper'
import Alerter from 'cozy-ui/transpiled/react/Alerter'
import { isReferencedByAlbum } from 'drive/web/modules/drive/files' // TODO move to cozy-client models
import { forceFileDownload } from 'cozy-stack-client/dist/utils'
import {
  saveFileWithCordova,
  saveAndOpenWithCordova
} from 'drive/mobile/lib/filesystem'

const { file: fileModel } = models
const { isDirectory } = fileModel

export const isAnyFileReferencedByAlbum = files => {
  for (let i = 0, l = files.length; i < l; ++i) {
    if (isReferencedByAlbum(files[i])) return true
  }
  return false
}

const isMissingFileError = error => error.status === 404

const downloadFileError = error => {
  return isMissingFileError(error)
    ? 'error.download_file.missing'
    : 'error.download_file.offline'
}

const openFileDownloadError = error => {
  return isMissingFileError(error)
    ? 'mobile.error.open_with.missing'
    : 'mobile.error.open_with.offline'
}

export const downloadFiles = async (client, files) => {
  if (files.length === 1 && !isDirectory(files[0])) {
    const file = files[0]

    try {
      const downloadURL = await client
        .collection('io.cozy.files')
        .getDownloadLinkById(file.id)
      const filename = file.name

      forceFileDownload(
        `${client.getStackClient().uri}${downloadURL}?Dl=1`,
        filename
      )
    } catch (error) {
      Alerter.error(downloadFileError(error))
      throw error
    }
  } else {
    const ids = files.map(f => f.id)
    const href = await client
      .collection('io.cozy.files')
      .getArchiveLinkByIds(ids)
    const fullpath = `${client.getStackClient().uri}${href}`
    forceFileDownload(fullpath, 'files.zip')
  }
}

const isAlreadyInTrash = err => {
  const reasons = err.reason !== undefined ? err.reason.errors : undefined
  if (reasons) {
    for (const reason of reasons) {
      if (reason.detail === 'File or directory is already in the trash') {
        return true
      }
    }
  }
  return false
}

export const trashFiles = async (client, files) => {
  try {
    for (const file of files) {
      await client.collection('io.cozy.files').destroy(file)
      client.collection('io.cozy.permissions').revokeSharingLink(file)
    }

    Alerter.success('alert.trash_file_success')
  } catch (err) {
    if (!isAlreadyInTrash(err)) {
      Alerter.error('alert.try_again')
    }
  }
}

export const exportFilesNative = async (client, files, filename) => {
  const downloadAllFiles = files.map(async file => {
    const response = await client
      .collection('io.cozy.files')
      .fetchFileContent(file)

    const blob = await response.blob()
    const filenameToUse = filename ? filename : file.name
    const localFile = await saveFileWithCordova(blob, filenameToUse)
    return localFile.nativeURL
  })

  try {
    Alerter.info('alert.preparing', {
      duration: Math.min(downloadAllFiles.length * 2000, 6000)
    })
    const urls = await Promise.all(downloadAllFiles)
    if (urls.length === 1 && isIOS()) {
      //TODO
      //It seems that files: is not well supported on iOS. url seems to work well
      //at with one file. Need to check when severals
      window.plugins.socialsharing.shareWithOptions(
        {
          url: urls[0]
        },
        result => {
          if (result.completed === true) {
            Alerter.success('mobile.download.success')
          }
        },
        error => {
          throw error
        }
      )
    } else {
      window.plugins.socialsharing.shareWithOptions(
        {
          files: urls
        },
        null,
        error => {
          throw error
        }
      )
    }
  } catch (error) {
    Alerter.error(downloadFileError(error))
  }
}

export const openFileWith = async (client, file) => {
  if (isMobileApp() && window.cordova.plugins.fileOpener2) {
    let fileData
    try {
      fileData = await client.collection('io.cozy.files').fetchFileContent(file)
    } catch (error) {
      Alerter.error(openFileDownloadError(error))
      throw error
    }

    const blob = await fileData.blob()
    try {
      await saveAndOpenWithCordova(blob, file.name)
    } catch (error) {
      Alerter.error('mobile.error.open_with.noapp')
    }
  } else {
    Alerter.error('mobile.error.open_with.noapp')
  }
}
