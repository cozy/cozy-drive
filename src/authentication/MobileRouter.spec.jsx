import { MobileRouter } from './MobileRouter'
import React from 'react'
import { shallow } from 'enzyme'
import { writeState, writeSecret } from './src/utils/onboarding'

describe('MobileRouter', () => {
  it('should render the appRoutes when all is well', () => {
    const app = shallow(
      <MobileRouter
        appRoutes={<div />}
        isAuthenticated={true}
        isRevoked={false}
        onboarding={{}}
        onboardingInformations={{}}
        history={{}}
        onAuthenticated={jest.fn()}
        onLogout={jest.fn()}
        appIcon={''}
      />
    )
    expect(app).toMatchSnapshot()
  })

  it('should render the appRoutes when no onboarding informations are present', () => {
    const app = shallow(
      <MobileRouter
        appRoutes={<div />}
        isAuthenticated={true}
        isRevoked={false}
        onboarding={{}}
        onboardingInformations={undefined}
        history={{}}
        onAuthenticated={jest.fn()}
        onLogout={jest.fn()}
        appIcon={''}
      />
    )
    expect(app).toMatchSnapshot()
  })

  it('should render the revoked view', () => {
    const app = shallow(
      <MobileRouter
        isAuthenticated={true}
        isRevoked={true}
        history={{}}
        onAuthenticated={jest.fn()}
        onLogout={jest.fn()}
        onboarding={{}}
        onboardingInformations={{}}
        appIcon={''}
        appRoutes={<div />}
      />
    )
    expect(app).toMatchSnapshot()
  })

  it('should render the auth screen when the onboarding has not started', () => {
    const onboardingInformations = {
      code: null,
      state: null,
      cozy_url: null
    }
    const app = shallow(
      <MobileRouter
        isAuthenticated={false}
        isRevoked={false}
        onboarding={{}}
        onboardingInformations={onboardingInformations}
        history={{}}
        onAuthenticated={jest.fn()}
        onLogout={jest.fn()}
        appIcon={''}
        appRoutes={<div />}
      />
    )
    expect(app).toMatchSnapshot()
  })

  it('should render nothing when onboarding', () => {
    const app = shallow(
      <MobileRouter
        client={jest.fn()}
        history={{}}
        isAuthenticated={false}
        isRevoked={false}
        onAuthenticated={jest.fn()}
        onLogout={jest.fn()}
        onboarding={{}}
        onboardingInformations={{ code: null }}
        appIcon={''}
        appRoutes={<div />}
      />
    )
    const instance = app.instance()
    const loginSpy = jest.spyOn(instance, 'doOnboardingLogin')
    const onboardingInformations = {
      code: '123',
      state: 'abc',
      cozy_url: 'http://lol.com'
    }

    app.setProps({
      onboardingInformations
    })

    expect(app).toMatchSnapshot()
    expect(loginSpy).toHaveBeenCalled()
  })

  describe('Auto Onboarding', () => {
    const onLogout = jest.fn()
    const onAuthenticated = jest.fn()
    const client = {
      stackClient: {
        fetchJSON: jest.fn(),
        fetch: jest.fn()
      }
    }
    const state = '123'
    const code = '111222333'
    const url = 'instance.cozy.test'
    const history = jest.fn()

    let app, instance

    beforeEach(() => {
      app = shallow(
        <MobileRouter
          client={client}
          history={{}}
          isAuthenticated={true}
          isRevoked={false}
          onAuthenticated={onAuthenticated}
          onLogout={onLogout}
          onboarding={{}}
          onboardingInformations={{}}
          appIcon={''}
          appRoutes={<div />}
        />
      )
      instance = app.instance()
    })

    afterEach(() => {
      jest.clearAllMocks()
    })

    it('should call logout when the local states are different', async () => {
      await writeState('456')

      await instance.doOnboardingLogin(state, code, url, history)
      expect(onLogout).toHaveBeenCalled()
    })

    it('should call logout when the exchanged secrets are different', async () => {
      await writeSecret('abc')
      await writeState('123')
      client.stackClient.fetchJSON.mockResolvedValue({
        onboarding_secret: 'def',
        onboarding_state: '456'
      })

      await instance.doOnboardingLogin(state, code, url, history)

      expect(client.stackClient.fetchJSON).toHaveBeenCalledTimes(1)
      expect(client.stackClient.fetchJSON.mock.calls[0][0]).toEqual('POST')
      expect(client.stackClient.fetchJSON.mock.calls[0][1]).toMatch(
        /\/auth\/secret_exchange$/
      )
      expect(onLogout).toHaveBeenCalled()
    })

    it('should call logout when the exchanged states are different', async () => {
      await writeSecret('abc')
      await writeState('123')
      client.stackClient.fetchJSON.mockResolvedValue({
        onboarding_secret: 'abc',
        onboarding_state: '456'
      })

      await instance.doOnboardingLogin(state, code, url, history)

      expect(client.stackClient.fetchJSON).toHaveBeenCalledTimes(1)
      expect(client.stackClient.fetchJSON.mock.calls[0][0]).toEqual('POST')
      expect(client.stackClient.fetchJSON.mock.calls[0][1]).toMatch(
        /\/auth\/secret_exchange$/
      )
      expect(onLogout).toHaveBeenCalled()
    })

    it('should fail when fetching the token produces an error', async () => {
      await writeSecret('abc')
      await writeState('123')
      client.stackClient.fetchJSON.mockResolvedValue({
        onboarding_secret: 'abc',
        onboarding_state: '123'
      })
      client.stackClient.fetch.mockRejectedValue({
        error: 'nope'
      })

      await instance.doOnboardingLogin(state, code, url, history)

      expect(client.stackClient.fetch).toHaveBeenCalledTimes(1)
      expect(client.stackClient.fetch.mock.calls[0][0]).toEqual('POST')
      expect(client.stackClient.fetch.mock.calls[0][1]).toMatch(
        /\/auth\/access_token/
      )
      expect(onLogout).toHaveBeenCalled()
    })

    it('should fail if the token cant be fetched', async () => {
      await writeSecret('abc')
      await writeState('123')
      client.stackClient.fetchJSON.mockResolvedValue({
        onboarding_secret: 'abc',
        onboarding_state: '123'
      })
      client.stackClient.fetch.mockResolvedValue({
        json: async () => {
          return {
            error: 'test error'
          }
        },
        status: 500
      })

      await instance.doOnboardingLogin(state, code, url, history)

      expect(client.stackClient.fetch).toHaveBeenCalledTimes(1)
      expect(client.stackClient.fetch.mock.calls[0][0]).toEqual('POST')
      expect(client.stackClient.fetch.mock.calls[0][1]).toMatch(
        /\/auth\/access_token/
      )
      expect(onLogout).toHaveBeenCalled()
    })

    it('should call onAuthenticated when everything worked', async () => {
      await writeSecret('abc')
      await writeState('123')
      client.stackClient.fetchJSON.mockResolvedValue({
        onboarding_secret: 'abc',
        onboarding_state: '123',
        clientInfo: 'test client'
      })
      client.stackClient.fetch.mockResolvedValue({
        json: async () => {
          return {
            error: null,
            value: 'secret token'
          }
        },
        status: 200
      })

      await instance.doOnboardingLogin(state, code, url, history)
      expect(onLogout).not.toHaveBeenCalled()
      expect(onAuthenticated).toHaveBeenCalledWith({
        url: 'https://' + url,
        token: {
          error: null,
          value: 'secret token'
        },
        clientInfo: {
          onboarding_secret: 'abc',
          onboarding_state: '123',
          clientInfo: 'test client'
        },
        router: history
      })
    })
  })
})
