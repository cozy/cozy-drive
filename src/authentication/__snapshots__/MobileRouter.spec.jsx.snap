// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileRouter should render nothing when onboarding 1`] = `""`;

exports[`MobileRouter should render the appRoutes when all is well 1`] = `
<Router
  history={Object {}}
  render={[Function]}
>
  <div />
</Router>
`;

exports[`MobileRouter should render the appRoutes when no onboarding informations are present 1`] = `
<Router
  history={Object {}}
  render={[Function]}
>
  <div />
</Router>
`;

exports[`MobileRouter should render the auth screen when the onboarding has not started 1`] = `
<Authentication
  appIcon=""
  onComplete={[MockFunction]}
  onException={[Function]}
  onboarding={Object {}}
  router={Object {}}
/>
`;

exports[`MobileRouter should render the revoked view 1`] = `
<Wrapper
  onLogBackIn={[MockFunction]}
  onLogout={[MockFunction]}
  router={Object {}}
/>
`;
